microsoft/PowerToys
# [ShortcutGuide] Registerhotkey interface

## Summary of the Pull Request

**What is this about:**
Centralized registering and population of hotkeys using `RegisterHotKey` API

**What is include in the PR:** 
- Extend `run_message_loop` to receive hotkeys
- Added `GetHotkeyEx` and `OnHotkeyEx` methods to PowerToysModuleInterface
- Added `CentralizedHotkeys` files to implement centralized registering and population
- Implemeted `get_config` in SG module interface. Current implementation requires valid settings

**How does someone test / validate:** 
- Verify if the shortcut brings up SG
- Verify if shortcuts can be changed from the settings page

## Quality Checklist


## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
